File: file/Sandig_CharlyHans/Python Version/MovieTime.py
from collections import deque

def main():
    movies = deque()
    snacks = deque()

    print("Enter 3 movies you would like to watch:")
    for _ in range(3):
        movie = input()
        movies.append(movie)

    print("Enter 3 snacks or beverages you would like to enjoy:")
    for _ in range(3):
        snack = input()
        snacks.append(snack)

    print("\nMovies to watch:")
    for movie in movies:
        print(movie)

    print("\nSnacks to enjoy:")
    for snack in snacks:
        print(snack)

    while snacks:
        print("\nType 'S' when done eating a snack:")
        input_text = input()
        if input_text.strip().upper() == 'S':
            snacks.popleft()  # Remove the first snack
            if snacks:
                print("Remaining snacks:")
                for remaining_snack in snacks:
                    print(remaining_snack)
            else:
                print("No more snacks.")

if __name__ == "__main__":
    main()

File: file/Sandig_CharlyHans/MovieTime/src/MovieTime.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class MovieTime {
    public static void main(String[] args) {
        Queue<String> movies = new LinkedList<>();
        Queue<String> snacks = new LinkedList<>();
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter 3 movies you would like to watch:");
        for (int i = 0; i < 3; i++) {
            String movie = scanner.nextLine();
            movies.add(movie);
        }

        System.out.println("Enter 3 snacks or beverages you would like to enjoy:");
        for (int i = 0; i < 3; i++) {
            String snack = scanner.nextLine();
            snacks.add(snack);
        }

        System.out.println("\nMovies to watch:");
        for (String movie : movies) {
            System.out.println(movie);
        }

        System.out.println("\nSnacks to enjoy:");
        for (String snack : snacks) {
            System.out.println(snack);
        }

        while (!snacks.isEmpty()) {
            System.out.println("\nType 'S' when done eating a snack:");
            String input = scanner.nextLine();
            if (input.equalsIgnoreCase("S")) {
                snacks.poll();
                if (!snacks.isEmpty()) {
                    System.out.println("Remaining snacks:");
                    for (String remainingSnack : snacks) {
                        System.out.println(remainingSnack);
                    }
                } else {
                    System.out.println("No more snacks.");
                }
            }
        }

        scanner.close();
    }
}
